### ISSUE TITLE
Games Results Returning Null Values For Developers

### ISSUE THREAD

**Issue by markusdrake** [kamranayub/GiantBomb-CSharp#16]
Hi Kamran,

Thanks for putting forth all this work and effort to build a quick plug and play C# API for us.

Right now I'm trying to troubleshoot my `GetGames()` call, which seems to be parsing most of the collections incorrectly and it results in null values. 

Collections coming back null:

- Developers
- Franchises
- Genres
- Images
- Publishers
- Releases
- SimilarGames

Collections that are okay:

- Platforms 

Any ideas what may be happening here? I did modify some of the models to include an Id number for my integration with Entity Framework. I'm wondering if the FastJsonDeserializer doesn't like that?

Side-note: I've noticed a lot of the dates under Platform coming back as values that SQL errors out on. Specifically [this type of issue](http://stackoverflow.com/questions/4608734/the-conversion-of-a-datetime2-data-type-to-a-datetime-data-type-resulted-in-an-o).

**Comment by kamranayub** [comment]
I'll run the test suite to see if anything is broken, I haven't noticed anything weird on my end (my background job calls the API every day). I would *definitely* suggest not using the models for data access--they should only be used for the API results. I would use separate entities and just map them (using AutoMapper or equivalent).

**Comment by markusdrake** [comment]
Thanks for the mapping idea, I could do that and keep all of it separate. I'll let you know when I get back to the office if my changes to the model was the cause of this problem. It may be all on my side.

**Comment by kamranayub** [comment]
I just ran the tests and it's all green, I explicitly ensure the collections aren't null when I get a single game. Sorry that doesn't help!

**Comment by markusdrake** [comment]
I re-worked everything so my data model is separate of the api model - just to make sure my changes weren't causing the bug. But I'm still getting empty results on some of the collections (like Developer).

[Here is a good example](http://www.giantbomb.com/api/game/3030-1/?api_key=YOUR_API_KEY_HERE&format=json) (Left my API Key out)

This is the very first game result in when making a query for `Games()`. 

The resulting model in my API request shows the following items are null:
- Developers
- Franchises
- Genres
- Images
- Publishers
- Releases
- SimilarGames

But data does exist for each one.

I'm going to pull the GiantBomb project down again as soon as I get a chance and confirm none of the files have been changed.

**Comment by markusdrake** [comment]
It's okay, every little bit helps. Thanks again.

**Comment by markusdrake** [comment]
Actually huge misunderstanding on my part - I think everything was working correctly I just need to make an additional call now to the specific `Game()` to get those other results in. Sorry about the troubles!

**Comment by kamranayub** [comment]
Yah, some fields are not present in list results... Super annoying, have to
fetch each game individually.

**Comment by markusdrake** [comment]
I was hitting the API limit pretty regularly once I needed game information, so I wrote a console app that uses HTMLAgilityPack to download the source of the page (site details url) and find the `platforms`, `developers`, `publishers`, and `genres` for each. I'm not downloading it all, but I calculated it would take this console app roughly 2.5 hours to get all the information & it doesn't tie up any of the API requests. 

My console app has a very specific use, since I'm tying it into my Entity Framework implementation, but if you think it's useful and want to look at it I'll send the code over to you. It might be a useful tool for somebody else.
